### ISSUE TITLE
Matrix events have a size limit; may interoperate badly with snapshots

### ISSUE THREAD

**Issue by reivilibre** [YousefED/Matrix-CRDT#11]
The title describes it all really, but Matrix events have a 65536 byte size limit (with the server-side attributes included), so this currently places an upper bound on the size of a snapshot (and thus presumably on the size of a document?) as they are encoded as events.

Do you have anything in mind to work around this; e.g. encoding snapshots as (encrypted) media?

**Comment by YousefED** [comment]
@reivilibre You're right. Have you run into this? The solution I have in mind would be fairly straightforward indeed; to send messages / snapshots that exceed the size limit as attachments (media). Afaik that approach shouldn't have any downsides

**Comment by vintprox** [comment]
> @reivilibre You're right. Have you run into this? The solution I have in mind would be fairly straightforward indeed; to send messages / snapshots that exceed the size limit as attachments (media). Afaik that approach shouldn't have any downsides

Ah, that's one way of surpassing that limit... as an option for size-intensive apps!

**Comment by ysds** [comment]
I ran into this problem with my own project as well.
Do you have a plan to tackle this problem? I really hope this problem will be solved.

**Comment by jchoi2x** [comment]
Ran into this issue as well.
